scripts/west_commands/runners/stlink_gdbserver.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) 2025 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Runner for debugging applications using the ST-LINK GDB server
+from STMicroelectronics, provided as part of the STM32CubeCLT.
+"""
+
+import argparse
+import platform
+import re
+import shutil
+from pathlib import Path
+
+from runners.core import RunnerCaps, RunnerConfig, ZephyrBinaryRunner
+
+STLINK_GDB_SERVER_DEFAULT_PORT = 61234
+
+
+class STLinkGDBServerRunner(ZephyrBinaryRunner):
+    @classmethod
+    def _get_stm32cubeclt_paths(cls) -> tuple[Path, Path]:
+        """
+        Returns a tuple of two elements of class pathlib.Path:
+            [0]: path to the ST-LINK_gdbserver executable
+            [1]: path to the "STM32CubeProgrammer/bin" folder
+        """
+
+        def find_highest_clt_version(tools_folder: Path) -> Path | None:
+            CUBECLT_FLDR_RE = re.compile("stm32cubeclt_([1-9]).([0-9]+).([0-9]+)", re.IGNORECASE)
+
+            def path2version(path: Path):
+                m = CUBECLT_FLDR_RE.match(path.name)
+                if m is None:
+                    # We should never be calling this function for
+                    # a path that did not match() the regexp, so
+                    # this should never happen...
+                    raise RuntimeError("Invalid candidate path?!")
+
+                major = int(m[1])
+                minor = int(m[2])
+                revis = int(m[3])
+                return major * 1000000 + minor * 1000 + revis
+
+            # List all possible CubeCLT installations in tools folder
+            print([k.name for k in tools_folder.iterdir()])
```

Comment:
Don't use `print`, is this useful to print each time anyway?

```suggestion
            # List all possible CubeCLT installations in tools folder
            self.logger.debug([k.name for k in tools_folder.iterdir()])
```

Reply:
Whoops, this is a leftover from debugging. We don't need this at all - removing it.